### DISCUSSION TITLE
How to simulate intermittent dialysis?

### DISCUSSION THREAD

**Discussion by yishuanwu** [Open-Systems-Pharmacology/Forum#845] [Questions & Problems]
I see the discussion for simulating renal impairment:
[How to simulate renal impair #479](https://github.com/Open-Systems-Pharmacology/Forum/issues/479)

However, how does one simulate intermittent hemodialysis in PK-Sim/MoBi?

**Comment by StephanSchaller** [comment]
You might use the same approach as this publication on adding external life support to a PK-Sim model :
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6202466/

Basically, you model the "external process" as a compartment/organ in MoBi with fluxes coming in and going out.
Additional processes (e.g. filtration) might then be added in this "organ".

Hope this helps,
Stephan

**Comment by yishuanwu** [comment]
I did suspected that I will need to add the external process and was worried that I might struggle with it. 
I am glad the paper shared the MoBi file, and I will try to see if I can make it work for dialysis instead of pulmonary ECMO. 

One part of my question is that I will have to figure out how to make the compartment available intermittently. 
I hypothesized that I can use time in the equation, so that I can code the timing of dialysis in the new container?
I am guessing MoBi uses C#, am I correct? I am planning to use remainder (it seems like the operator is %) and nested if else statement to determine the timing of dialysis in the container.

**Comment by StephanSchaller** [comment]
MoBi is GUI-based.

It allows a number of math expressions (see [here](https://docs.open-systems-pharmacology.org/working-with-mobi/mobi-documentation/model-building-components#working-with-formulas)).

There you could use a time-dependent formula (directly in the flow formula or with an Event) to switch flow on or off, depending on when dialysis is connected (similar to an administration protocol).

Hope this helps,
Stephan

**Comment by yishuanwu** [comment]
The list of formula does not have a function for calculating the remainder in the list. Does that mean that it is not capable of calculating the remainder? 
The reason I want to calculate the remainder is that dialysis is usually on a week based schedule, e.g. MWF. I was thinking to get the remainder from dividing the time by 168 h to get 24-28 h, 72-76 h, and 120-124 h. 
But you mentioned the word event. It seems like it may be a better approach.

For me, tissue distribution rather than the clearance is the focus. Would it be reasonable to "clamp" the renal and metabolism processes to reduce clearance at all times except the dialysis event?

**Comment by yishuanwu** [comment]
Here's my progress so far:
1, Added a Dialysis container similar to the ECMO example, with blood cells, plasma, and dialysate waste compartments
2. Added passive transport processes which is zero when dialysis is not active
3. Created dialysis events similar to the gallbladder processes

I got it to work. Will share the details later.
